src/hooks/useWalletConnect.ts

```diff
@@ -166,19 +79,40 @@ export default function useWalletConnect() {
 
   const accounts = useAccounts(client);
 
-  const addPendingFlow = usePendingFlowStore(
-    pendingFlowSelector.addPendingFlow
-  );
-  const clearPendingFlow = usePendingFlowStore(
-    pendingFlowSelector.clearPendingFlow
+  const proposalParams = useParams({ from: "/proposal/$id" });
+
+  const onProposalExpire = useCallback(
+    (proposalExpire: Web3WalletTypes.ProposalExpire) => {
+      void queryClient
+        .invalidateQueries({
+          queryKey: pendingSessionsProposalsQueryKey,
+        })
+        .then(() => {
+          // TODO maybe remove and instead find proposal in the array from usePendingSessionsProposals on proposal route
```

Comment:
👍 good idea

Reply:
It's what is done for the session detail page, but I first tried to use the store of wallet connect directly to do the get proposal in the proposal route
In the end it feels simpler to not use the get from wallet-connect as it's easier to invalidate cache with react-query all around the app